phpBB/phpbb/log/log.php

```diff
@@ -521,15 +521,73 @@ public function get_logs($mode, $count_logs = true, $limit = 0, $offset = 0, $fo
 			$sql_keywords = $this->generate_sql_keyword($keywords);
 		}
 
-		if ($count_logs)
-		{
-			$sql = 'SELECT COUNT(l.log_id) AS total_entries
-				FROM ' . $this->log_table . ' l, ' . USERS_TABLE . ' u
-				WHERE l.log_type = ' . (int) $log_type . '
-					AND l.user_id = u.user_id
+		$get_logs_sql_ary = array(
+			'SELECT' => 'l.*, u.username, u.username_clean, u.user_colour',
+			'FROM' => $this->log_table . ' l, ' . USERS_TABLE . ' u',
+			'WHERE' => 'l.user_id = u.user_id
 					AND l.log_time >= ' . (int) $log_time . "
 					$sql_keywords
-					$sql_additional";
+					$sql_additional",
+
+			'ORDER_BY' => $sort_by,
+		);
+
+		if($log_type){
+			$get_logs_sql_ary['WHERE'] = 'l.log_type = ' . (int) $log_type . '
+					AND ' . $get_logs_sql_ary['WHERE'];
+		}
+
+		/**
+		* Modify the query to obtain the logs data
+		*
+		* @event core.get_logs_main_query_before
```

Comment:
Ah didn't see the log_type validation there :/ hmm yeah I guess then 2 events are required.
Then its fine I guess we need two events then